src/main/java/ac/grim/grimac/player/GrimPlayer.java

```diff
@@ -504,6 +504,8 @@ public void updatePermissions() {
         if (bukkitPlayer == null) return;
         this.noModifyPacketPermission = bukkitPlayer.hasPermission("grim.nomodifypacket");
         this.noSetbackPermission = bukkitPlayer.hasPermission("grim.nosetback");
+        FoliaScheduler.getAsyncScheduler().runNow(GrimAPI.INSTANCE.getPlugin(), t-> checkManager.allChecks.values().stream().map(c->(Check)c)
+                .forEach(Check::updateExempted));
```

Comment:
Do not cast without checking
```suggestion
        FoliaScheduler.getAsyncScheduler().runNow(GrimAPI.INSTANCE.getPlugin(), t ->
            for (AbstractCheck check : checkManager.allChecks.values()) {
                if (check instanceof Check) {
                    ((Check) check).updateExempted();
                }
            }
        );
```